src/qibolab/pulses.py

```diff
@@ -1181,7 +1181,7 @@ def plot(self, savefig_filename=None):
         import matplotlib.pyplot as plt
         from matplotlib import gridspec
 
-        num_samples = int(np.rint(self.duration / 1e9 * PulseShape.SAMPLING_RATE))
+        num_samples = len(self.shape.envelope_waveform_i.data)
```

Comment:
Since part of the task is already performed somewhere else:
https://github.com/qiboteam/qibolab/blob/5f7819d5b0e8436f4a316a9c5010abc14271bdbf/src/qibolab/pulses.py#L46-L49

you could even slightly simplify this call
```suggestion
        num_samples = len(self.shape.envelope_waveform_i)
```